ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -137,10 +163,104 @@ public static Bytes32 getAttestationMessageToSign(AttestationData attestationDat
+    List<UnsignedLong>  custody_bit_1_indices = get_attesting_indices(state, attestation.getData(), attestation.getCustody_bitfield());
+
+    List<UnsignedLong>  custody_bit_0_indices = new ArrayList<UnsignedLong>();
+    for(int i = 0; i < attestations.size(); i++){
+      if(!custody_bit_1_indices.contains(UnsignedLong.valueOf(i)))custody_bit_0_indices.add(UnsignedLong.valueOf(i));
+    }
+    return new IndexedAttestation(
+            custody_bit_0_indices,
+            custody_bit_1_indices,
+            attestation.getData(),
+            attestation.getAggregate_signature()
+            );
+  }
+
+  public static List<UnsignedLong> get_attesting_indices(
+          BeaconState state, AttestationData attestation_data, Bytes bitfield) throws IllegalArgumentException {
+    //Return the sorted attesting indices corresponding to ``attestation_data`` and ``bitfield``.
+    List<Integer> committee = get_crosslink_committee(state, attestation_data.getTarget_epoch(), attestation_data.getCrosslink().getShard());
+    checkArgument(verify_bitfield(bitfield, committee.size()));
+    Iterator<Integer> itr = committee.iterator();
+
+    List<UnsignedLong> returnValue = new ArrayList<UnsignedLong>();
+    while(itr.hasNext()){
+      int index = itr.next().intValue();
+      int bitfield1 = get_bitfield_bit(bitfield, index);
+      if((get_bitfield_bit(bitfield, index) & 1) == 1) returnValue.add(UnsignedLong.valueOf(index));
+    }
+    return returnValue;
+  }
+
+  public static void validate_indexed_attestation(BeaconState state, IndexedAttestation indexed_attestation){
+    //Verify validity of ``indexed_attestation``.
+    List<UnsignedLong> bit_0_indices = indexed_attestation.getCustody_bit_0_indices();
+    List<UnsignedLong> bit_1_indices = indexed_attestation.getCustody_bit_1_indices();
+
+    //Verify no index has custody bit equal to 1 [to be removed in phase 1]
+    checkArgument((bit_0_indices.size() == 0);
+    //Verify max number of indices
+    checkArgument(bit_0_indices.size() + bit_1_indices.size() <= MAX_INDICES_PER_ATTESTATION);
+    //Verify index sets are disjoint
+    checkArgument(intersection(bit_0_indices, bit_1_indices).size() == 0);
+    //Verify indices are sorted
+    Collections.sort(bit_0_indices);
+    List<UnsignedLong> bit_0_indices_sorted = new ArrayList<UnsignedLong>(bit_0_indices);
+    Collections.sort(bit_0_indices_sorted);
+    List<UnsignedLong> bit_1_indices_sorted = new ArrayList<UnsignedLong>(bit_1_indices);
+    Collections.sort(bit_1_indices_sorted);
+    checkArgument(bit_0_indices.equals(bit_0_indices_sorted) && bit_1_indices.equals(bit_1_indices_sorted));
+
+    List<Validator> validators = state.getValidator_registry();
+    List<BLSPublicKey> pubkeys = Arrays.stream(validators.subList(0, bit_0_indices.size()).toArray()).map((validator) -> {return ((Validator)validator).getPubkey();}).collect(Collectors.toList());
```

Comment:
Here instead of getting the sublist, we take each Validator in Validator registry that bit_0_indices map to using bit_0_indices.map(i -> state.getValidator_registry().get(i)) and then we call bls_aggregate_pubkeys on this. same with bit_1_indices. But we shouldnt be adding all those validators to the same list.